h2o-core/src/main/java/hex/GainsLift.java

```diff
@@ -167,6 +167,8 @@ public TwoDimTable createTwoDimTable() {
       sum_s_i += n_i * s_i;
       double lift=p_i/P; //can be NaN if P==0
       double sum_lift=(double)sum_e_i/sum_n_i/P; //can be NaN if P==0
+      double cum_event = sum_e_i/(double)E;
+      double cum_non_event = (sum_n_i - sum_e_i) / (double) (N - E);
```

Comment:
avg_response_rate can be 1 and thus N can be equal to E, leading to a division by zero

Reply:
I don't see how this is possible. The `init` method has:

```
    if (_labels.cardinality() != -1 && _labels.cardinality() != 2)
      throw new IllegalArgumentException("Actual column must contain binary class labels, but found cardinality " + _labels.cardinality() + "!");
```

Which makes sure there is at least one non-response (or vice versa) in the actual labels. This hould result in 0.9999 rate but not `1`.

Reply:
I've added a test to demonstrate this behavior into `GainsLiftTest.java`.

Reply:
I can see some rounding errors could result in the resp. rate to be 1. I've put a guard against that into the code:

```
      final double cum_event = sum_e_i / (double)E;
      final double total_non_event = (double) (N - E);
      // If response rate is 1, there are non non-events and the cumulative count will always be zero
      final double cum_non_event = total_non_event == 0 ? 0 : (sum_n_i - sum_e_i) / total_non_event;
```

Reply:
Yes, but there are various edge cases that this check doesn't cover. You can eg. have zero weights for one of the predicted classes and the check will not catch that. See here: https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/main/java/hex/GainsLift.java#L238

Reply:
A new test `testAverageResponseRate` has been added to test exactly that case. Resolved.